### DISCUSSION TITLE
Client device for ServerInertial

### DISCUSSION THREAD

**Discussion by traversaro** [robotology/community#80] [Q&A] [answered]
The yarp devices permit to access a remote device on a computer across the network using a pair of devices : the `wrapper` is attached to the remote `wrapped` device and is used to expose the device functionality over a set of yarp ports , while the `client` is used to connect to the wrapper and provide the same C++ interface of the original device, see http://www.yarp.it/note_devices.html . 

We have instantiations of this pattern for: 
- ControlBoards devices : 
  - `wrapped` : [CanBusMotionControl](http://wiki.icub.org/brain/classyarp_1_1dev_1_1CanBusMotionControl.html) , [embObjMotionControl](http://wiki.icub.org/brain/classyarp_1_1dev_1_1embObjMotionControl.html)
  - `wrapper` :  [ControlBoardWrapper](http://www.yarp.it/classyarp_1_1dev_1_1ControlBoardWrapper.html)
  - `client` : [RemoteControlBoard](http://www.yarp.it/classyarp_1_1dev_1_1RemoteControlBoard.html)
- Analog (i.e. sensor publishing a vector of floating point numbers): 
  - `wrapped` : [CanBusAnalogSensor](http://wiki.icub.org/brain/classCanBusAnalogSensor.html) , [embObjAnalogSensor](http://wiki.icub.org/iCub_documentation/classyarp_1_1dev_1_1embObjAnalogSensor.html)
  - `wrapper` :  [AnalogWrapper](http://www.yarp.it/classyarp_1_1dev_1_1AnalogWrapper.html)
  - `client` : [AnalogSensorClient](http://www.yarp.it/classyarp_1_1dev_1_1AnalogSensorClient.html)
- Inertial Measurement Units-like devices (implementing `IGenericSensor` interface) : 
  - `wrapped` : [XSensMTx](http://wiki.icub.org/brain/classyarp_1_1dev_1_1XSensMTx.html)
  - `wrapper` : [ServerInertial](http://www.yarp.it/classyarp_1_1dev_1_1ServerInertial.html)
  - `client` : ? 

My question is: does a `client` device that connects to ports opened by a `ServerInertial` exists? 
@barbalberto

**Comment by barbalberto** [comment, marked as answer]
Yes, **it is missing since ever.** Right now **`AnalogSensorClient` can be used** for this purpose.
When I've free time, I'd like to fix this type of dis-homogeneity and this one is in the list since long.
IMHO for this device it would be also better to define a _structured_ type instead of simply an array of data, just to easily distinguish between acceleration, Euler angles and so on.

**Comment by pattacini** [comment]
@barbalberto if you want to take on this development I think it'd be worth tracking it with proper tickets on the corresponding repositories.
Closing.
